pkg/operator/clustermembercontroller/clustermembercontroller.go

```diff
@@ -35,27 +38,40 @@ type ClusterMemberController struct {
 	podLister      corev1listers.PodLister
 	nodeLister     corev1listers.NodeLister
 	networkLister  configv1listers.NetworkLister
+	// machineAPIChecker determines if the precondition for this controller is met,
+	// this controller can be run only on a cluster that exposes a functional Machine API
+	machineAPIChecker ceohelpers.MachineAPIChecker
+
+	masterMachineLister   machinelistersv1beta1.MachineLister
+	masterMachineSelector labels.Selector
 }
 
 func NewClusterMemberController(
```

Comment:
changes for this controller appear safe regardless, let's make this PR smaller by splitting these out.

Reply:
done, https://github.com/openshift/cluster-etcd-operator/pull/780